src/main/java/com/nvidia/spark/rapids/jni/ParseURI.java

```diff
@@ -30,76 +30,106 @@ public class ParseURI {
   /**
    * Parse protocol for each URI from the incoming column.
    *
-   * @param URIColumn The input strings column in which each row contains a URI.
+   * @param uriColumn The input strings column in which each row contains a URI.
+   * @param failOnError If true, use ANSI-aware parsing.
    * @return A string column with protocol data extracted.
    */
-  public static ColumnVector parseURIProtocol(ColumnView uriColumn) {
+  public static ColumnVector parseURIProtocol(ColumnView uriColumn, boolean failOnError) {
     assert uriColumn.getType().equals(DType.STRING) : "Input type must be String";
-    return new ColumnVector(parseProtocol(uriColumn.getNativeView()));
+    if (failOnError) {
+      return new ColumnVector(parseProtocol(uriColumn.getNativeView(), true));
+    } else {
+      return new ColumnVector(parseProtocol(uriColumn.getNativeView(), false));
+    }
```

Comment:
```suggestion
    return new ColumnVector(parseProtocol(uriColumn.getNativeView(), failOnError));
```

Reply:
Done here and in all the relevant places. PTAL.